packages/vaadin-grid/src/vaadin-grid.js

```diff
@@ -40,8 +40,7 @@ const TOUCH_DEVICE = (() => {
 
```

Comment:
Not sure why but this page just renders an empty grid (works on master):
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script type="module">
      import '@vaadin/vaadin-grid/all-imports';
      import '@vaadin/vaadin-template-renderer';

      const grid = document.querySelector('vaadin-grid');
      grid.items = [{ name: 'foo' }, { name: 'bar' }];
    </script>

    <vaadin-grid>
      <vaadin-grid-column>
        <template class="header">Header</template>
        <template>[[item.name]]</template>
        <template class="footer">Footer</template>
      </vaadin-grid-column>
    </vaadin-grid>
  </body>
</html>
```

Reply:
Oh, the actual issue is that `vaadin-grid` is imported before `vaadin-template-renderer` that means `window.Vaadin.templateRendererCallback` is not available yet at that point.

Possible solutions I see:

### Solution 1

Create something like a queue of callbacks that is supposed to be flushed by the template renderer when it is ready.

Instead of:
```js
if (window.Vaadin.templateRendererCallback) {
  window.Vaadin.templateRendererCallback(component);
}
```

Do this:
```js
connectedCallback() {
  window.Vaadin.templateRendererQueue = window.Vaadin.templateRendererQueue || [];
  window.Vaadin.templateRendererQueue.push(this);
}

disconnectedCallback() {
  window.Vaadin.templateRendererQueue = window.Vaadin.templateRendererQueue || [];
  window.Vaadin.templateRendererQueue = window.Vaadin.templateRendererQueue.filter((component) => {
    return component !== this;
  });
}
```

### Solution 2

Invoke the template renderer callback later, but it likely will increase the number of renderers' calls.

```js
nextRender(() => {
  if (window.Vaadin.templateRendererCallback) {
    window.Vaadin.templateRendererCallback(component);
  }
});
```

### Solution 3

Recommend importing the template renderer before all other components, but personally, I don't think it is a reliable solution.

---

What do you think?

Reply:
Ah, right. I think we agreed with 3 initially. Let's not change that in scope of this PR. Maybe we could consider 1 (or 2) later.

But even if I change the import order on the example page, the grid doesn't render properly. Might be related to https://github.com/vaadin/web-components/pull/2052#discussion_r647458418

Reply:
Yeah, seems there are 2 issues here at the same time. When I don't import grids in the template renderer and import the template renderer before grids here, then everything works properly.

Reply:
We could add marker properties such as:
```js
/** @private */
__gridColumnElement = true;
```
that can be used by template-renderer for identifying the types without having to depend on/import grid.

Reply:
Great idea! 👍  Pushed a commit according to the discussion.